lib/protocol/stream.js

```diff
@@ -273,7 +279,24 @@ Stream.prototype._receive = function _receive(frame, ready) {
   // * If it's a DATA frame, then push the payload into the output buffer on the other side.
   //   Call ready when the other side is ready to receive more.
   if (!this._ended && (frame.type === 'DATA')) {
-    var moreNeeded = this.push(frame.data);
+    if (this.contentEncoding === 'gzip' && frame.data.length > 0)
+    {
+      if (!this.pakoInf)
+      {
+        var self = this;
+        this.pakoInf = new pako.Inflate();
+        this.pakoInf.onData = function(data)
+        {
+            self.push(Buffer.from(data));
```

Comment:
don't merge.  We should be able to look at the frame to see if it is an end frame and then explicitly pass true/false on final frame to this.pakoInf.push(data, finished)